### DISCUSSION TITLE
Inherit operators support for ConversionOperatorMapper

### DISCUSSION THREAD

**Discussion by blemasle** [LuckyPennySoftware/AutoMapper#3722] [Ideas]
Hi,

I was wondering if there was any reason not to use `FlattenHierarchy` rather than `DeclaredOnly` in [`ConversionOperatorMapper`](https://github.com/AutoMapper/AutoMapper/blob/master/src/AutoMapper/Mappers/ConversionOperatorMapper.cs#L13) to support inherited operators ?

For some simple use cases, not being able to automatically use inherited operators is a bit of a bummer.

(oversimplified example)

```csharp
public class IntValue
{
    public IntValue(int value)
    {
        Value = value;
    }

    public int Value { get; }

    public static explicit operator int(IntValue intValue)
        => intValue.Value;
}

public class SomethingId : IntValue
{
    public SomethingId(int value)
        : base(value)
    { }
}

```

```csharp
var mapper = new MapperConfiguration(_ => {}).CreateMapper();
mapper.Map<int>(new SomethingId(3)); // fails because ConversionOperatorMapper does not find the inherited explicit operator
```

Regards

**Comment by lbargaoanu** [comment]
Perhaps it simply didn't come up before :) You can PR my branch.

**Reply by blemasle** [reply]
Done. See https://github.com/AutoMapper/AutoMapper/pull/3723 🙂

**Reply by lbargaoanu** [reply]
Thanks for the PR! Try the [MyGet](https://docs.automapper.org/en/latest/The-MyGet-build.html) build.

**Reply by lbargaoanu** [reply]
We might have a problem when both the base class and the derived class define conversion operators. I guess we'll see...

**Reply by blemasle** [reply]
Using `FlattenHierarchy` returns the members from the derived types first, so it should also work as expected in that case. But it is very dependent on the dotnet implementation 😕 

The MyGet build solves the issue we had, as expected.
Thanks, and sorry for the corrections you had to make on such a small PR.

**Reply by lbargaoanu** [reply]
No worries, it just seemed easier to make the changes and merge :)
